xmtp_mls/src/storage/encrypted_store/user_preferences.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{impl_store, storage::StorageError, Store};
+
+use super::{
+    schema::user_preferences::{self, dsl},
+    DbConnection,
+};
+use diesel::prelude::*;
+
+#[derive(Identifiable, Insertable, Queryable, Debug, Clone, PartialEq, Eq, Default)]
+#[diesel(table_name = user_preferences)]
+#[diesel(primary_key(id))]
+pub struct StoredUserPreferences {
+    /// Primary key - latest key is the "current" preference
+    pub id: Option<i32>,
+    /// Randomly generated hmac key root
+    pub hmac_key: Option<Vec<u8>>,
+}
+impl_store!(StoredUserPreferences, user_preferences);
+
+impl StoredUserPreferences {
+    pub fn load(conn: &DbConnection) -> Result<Self, StorageError> {
```

Comment:
Suggestion: maybe the fetch macro and the insert macro could be used here, otherwise looks good.